apm-sniffer/apm-sdk-plugin/mongodb-2.x-plugin/src/test/java/org/skywalking/apm/plugin/mongodb/v2/MongoDBV2MethodInterceptorTest.java

```diff
@@ -0,0 +1,125 @@
+import static org.mockito.Mockito.when;
+import static org.skywalking.apm.agent.test.tools.SpanAssert.assertException;
+
+@RunWith(PowerMockRunner.class)
+@PowerMockRunnerDelegate(TracingSegmentRunner.class)
+public class MongoDBV2MethodInterceptorTest {
+
+    @SegmentStoragePoint
+    private SegmentStorage segmentStorage;
+
+    @Rule
+    public AgentServiceRule serviceRule = new AgentServiceRule();
+
+    private MongoDBV2MethodInterceptor interceptor;
+
+    @Mock
+    private EnhancedInstance enhancedInstance;
+
+    private Object[] arguments;
+    private Class[] argumentTypes;
+
+    @SuppressWarnings({"rawtypes", "unchecked"})
+    @Before
+    public void setUp() throws Exception {
+
+        interceptor = new MongoDBV2MethodInterceptor();
+
+        Config.Plugin.MongoDB.TRACE_PARAM = true;
+
+        when(enhancedInstance.getSkyWalkingDynamicField()).thenReturn("127.0.0.1:27017");
+
+    }
+
+    @Test
+    public void testIntercept() throws Throwable {
+        interceptor.beforeMethod(enhancedInstance, getExecuteMethod(), null, null, null);
+        interceptor.afterMethod(enhancedInstance, getExecuteMethod(), null, null, null);
+
+        MatcherAssert.assertThat(segmentStorage.getTraceSegments().size(), is(1));
+        TraceSegment traceSegment = segmentStorage.getTraceSegments().get(0);
+        List<AbstractTracingSpan> spans = SegmentHelper.getSpans(traceSegment);
+        assertMongoSpan(spans.get(0));
+    }
+
+    @Test
+    public void testInterceptWithException() throws Throwable {
+        interceptor.beforeMethod(enhancedInstance, getExecuteMethod(), null, null, null);
+        interceptor.handleMethodException(enhancedInstance, getExecuteMethod(), null, null, new RuntimeException());
+        interceptor.afterMethod(enhancedInstance, getExecuteMethod(), null, null, null);
+
+        MatcherAssert.assertThat(segmentStorage.getTraceSegments().size(), is(1));
+        TraceSegment traceSegment = segmentStorage.getTraceSegments().get(0);
+        List<AbstractTracingSpan> spans = SegmentHelper.getSpans(traceSegment);
+        assertMongoSpan(spans.get(0));
+        List<LogDataEntity> logDataEntities = SpanHelper.getLogs(spans.get(0));
+        assertThat(logDataEntities.size(), is(1));
+        assertException(logDataEntities.get(0), RuntimeException.class);
+    }
+
+    private void assertMongoSpan(AbstractTracingSpan span) {
+        assertThat(span.getOperationName(), is("MongoDB/getWriteConcern"));
+        assertThat(SpanHelper.getComponentId(span), is(9));
+        List<KeyValuePair> tags = SpanHelper.getTags(span);
+        assertThat(tags.get(0).getValue(), is("MongoDB"));
+        assertThat(span.isExit(), is(true));
+        assertThat(SpanHelper.getLayer(span), is(SpanLayer.DB));
+    }
+
+    private Method getExecuteMethod() {
+        try {
+            return Mongo.class.getMethod("getWriteConcern");
+        } catch (NoSuchMethodException e) {
+            e.printStackTrace();
```

Comment:
I think `e.printStackTrace()` should be remove.

Reply:
@ascrutae  Yes, I have modified it